app/views/assignments/_read.html.erb

```diff
@@ -40,7 +40,8 @@
     <p><%= @assignment.submission_rule.class.human_attribute_name(:description) unless @assignment.is_timed %></p>
 
     <ul>
-      <% if @grouping.nil? %>
+      <% if @grouping.nil? && @assignment.group_max > 1 &&
```

Comment:
Ah, I realized here that the old check (just `@grouping.nil?`) prevented the rest of the conditions/branches below from executing when `@grouping` is `nil`. Otherwise, for example, the `@grouping.start_time.nil?` check below raises an error when `@grouping` is `nil`.

Since the purpose is to only display the message when the student needs to create a group, you should use a nested `if` here, and leave the original condition unchanged.